hibernate-core/src/main/java/org/hibernate/spi/NavigablePath.java

```diff
@@ -140,9 +140,10 @@ public boolean equals(Object other) {
 			if ( ! Objects.equals( getAlias(), otherNavigablePath.getAlias() ) ) {
 				return false;
 			}
+			return Objects.equals( getRealParent(), otherNavigablePath.getRealParent() );
```

Comment:
I have just done the same change for https://github.com/hibernate/hibernate-orm/pull/5140/files :)